rcl_logging_spdlog/src/rcl_logging_spdlog.cpp

```diff
@@ -0,0 +1,174 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cerrno>
+#include <memory>
+#include <mutex>
+#include <inttypes.h>
+
+#include <rcutils/allocator.h>
+#include <rcutils/filesystem.h>
+#include <rcutils/get_env.h>
+#include <rcutils/process.h>
+#include <rcutils/snprintf.h>
+#include <rcutils/time.h>
+
+#include "spdlog/spdlog.h"
+#include "spdlog/sinks/basic_file_sink.h"
+
+#include "rcl_logging_spdlog/logging_interface.h"
+
+#define RC_LOGGING_RET_OK                          (0)
+#define RC_LOGGING_RET_ERROR                       (2)
```

Comment:
Are the parens necessary? Also the white space seems excessive.

Reply:
They are not strictly necessary, but are usually recommended when doing macros like this.  I'll fix up the whitespace.